opencog/benchmark/diary.txt

```diff
@@ -816,3 +816,117 @@ Observations:
 
 -- addLink is faster for the scheme, than the native C++.  ... why?
    some bizarre cache contention ???
+
+==================================================================
+15 March 2015:
+
+by Curtis Faith
+
+Baseline test before change from boost::shared_mutext to boost::mutex 
+in ClassServer::isA(Type sub, Type super) which I tested after doing
+research, including this insightful post:
+
+http://permalink.gmane.org/gmane.comp.lib.boost.devel/211180
+
+by:
+
+http://www.boost.org/users/people/anthony_williams.html
+
+Benchamrks performed on 2014 MacBook Air:
+   Intel Dual-Core i5
+   1.4 GHz
+   256 KB L2 cache
+   3 MB L3 cache
+   4GB 1600 Mhz DDR3 RAM
+
+Benchmarking AtomSpace's getHandleSet method 100000 times ..........
+122.537761 seconds elapsed (816.07 per second)
+Sum clock() time for all requests: 116957317 (116.957 seconds, 855.013 requests
+per second)
+
+after change to boost::mutex
+
+Benchmarking AtomSpace's getHandleSet method 100000 times ..........
+116.396391 seconds elapsed (859.13 per second)
+Sum clock() time for all requests: 110356511 (110.357 seconds, 906.154 requests
+per second)
+------------------------------
+
+This results in a 6.5% increase in throughput.
+
+==================================================================
+15 March 2015:
+
+by Curtis Faith
+
+Noticed that the result set which is being back_inserted did not do a reserve
+for the size ahead of the insertions and most of the time is being spent
+copying and initializing. So I added a getNumAtomsOfType function to TypeIndex
+so that the getHandleSet test could call reserve on the result handle before
```

Comment:
In the code that I'm re-doing, I added a function size(Type)  that returns the size but I cannot figure out wehre you wanted to call this.